src/user/routes.rs

```diff
@@ -0,0 +1,69 @@
+use crate::user::{User, UserRequest};
+
+use actix_web::{delete, get, post, put, web, HttpResponse, Responder};
+use sqlx::postgres::PgPool;
+
+#[get("/users/")]
+async fn find_all(db_pool: web::Data<PgPool>) -> impl Responder {
+    let result = User::find_all(db_pool.get_ref()).await;
+    match result {
+        Ok(users) => HttpResponse::Ok().json(users),
+        _ => HttpResponse::BadRequest().body("Error trying to read all users from database"),
+    }
+}
+
+#[get("/user/{id}/")]
+async fn find(id: web::Path<i32>, db_pool: web::Data<PgPool>) -> impl Responder {
+    let result = User::find_by_id(id.into_inner(), db_pool.get_ref()).await;
```

Comment:
You can pattern match in the parameters, allowing you to do `web::Path(id): web::Path<i32>` and get rid of `id.into_inner()`. This is the case with `web::Data`, `web::Json`, and other extracted parameters in handlers.